### PR TITLE
Create member count method as currently there is no way to get the

### PR DESCRIPTION
**Pull request by csteifel**
underlying cached member count quickly, you would have to use iterators.

### CODE REVIEW

File: rabbit.hpp [hunk 1]
```diff
@@ -735,6 +735,13 @@ class basic_value_ref
   value_ref_type operator[](const string_ref_type& name) { return at(name); }
   const_value_ref_type operator[](const string_ref_type& name) const { return at(name); }
 
+
+  size_t member_count() const
+  {
+    type_check<object_tag>();
+    return value_->MemberCount();
+  }
```
Comment (mashiro) [inline rabbit.hpp:743]: It would be simpler to branch by tag with size method.

File: test/array_test.cpp [hunk 2]
```diff
@@ -5,7 +5,7 @@
 BOOST_AUTO_TEST_CASE(size_test)
 {
   rabbit::value v;
-  BOOST_CHECK_THROW(v.size(), rabbit::type_mismatch);
+  BOOST_CHECK_THROW(v.size(), std::runtime_error);
```
Comment (mashiro) [inline test/array_test.cpp:8]: Why change type_mismatch to runtime_error?

Comment (csteifel) [reply test/array_test.cpp:8]: I felt as though it wasn't a properly descriptive exception type anymore. To me mismatch implies that there is only 1 type that would actually match, when in fact now there are two types that could be valid here. I'm not too attached to it to not make changes though, I can prep a change back to type_mismatch.

Comment (mashiro) [reply test/array_test.cpp]: I also feel that the name type_mismatch is not good. But please use this currently.

**Pull request merged**